app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,27 @@
+class SessionsController < ApplicationController
+
+  def new
+  end
+
+  def create
+    user = User.find_by(username: params[:session][:email]) || User.find_by(email: params[:session][:email])
```

Comment:
I'm pretty sure you can just make this argument session_params because of your private method right?